home/rvagg/2021-02-exploration-report-small-filecoin-deals.md

```diff
@@ -0,0 +1,134 @@
+    * Notaries are organisations that can hand out "DataCap" to clients
+    * Notaries are selected by the Filecoin community, distributed around the world
+    * "DataCap" is a quota that can be spent, along with FIL, to make verified deals
+    * Verified deals contribute 10x to miner power, proportional to how much space they occupy in a sector
+    * Therefore, verified deals should be attractive to miners who may charge less FIL to get them
+
+### Deal incentives & challenges
+
+* Miners bear responsibility for publishing deals to the chain, with high gas costs this can be in the order of $5
+* Recent changes allow for batch publishing deals, but anecdotally this hasn't changed the economics for miners much (need more context here - is it because the batch publish is still quite expensive, or because there are not enough deals to batch?)
+* Accepting deals can offset the cost of sealing and proving, but this biases toward larger deals
+* Verified deals have the 10x increase in power, but they are scarce because FIL+ is so new, currently unknown the scale of incentive impact for miners for large and small deals
+* Bugs in Lotus have made miners gun-shy on doing anything more than CC:
+  * Transferring deal data is buggy and transfers often fail (anecdotal)
+  * Retrieval is buggy and can cause crashes, which are problematic when you need 100% uptime to prove sectors (anecdotal)
+* A lot of large miners don't seem to have transitioned from CC sealing to accepting deals - could be immaturity of market
+* Many miners don't respond to `query-ask` at all. Of those that do, the majority appear to reject deals even when those deals fit within the ask parameters. So the `query-ask` mechanism seems of limited use in its current form.
+
+Filecoin currently only stores ~5 Pb of deal data and growth rate is not large.
+
+## Installing and Running Lotus
+
+Followed instructions at https://docs.filecoin.io/get-started/lotus/installation/ for an AWS instance
+
+ * Large AMD EPYC instance, Ubuntu 20.04
+ * Installed dependencies (used Go Snap rather than Go tarball)
+ * Cloned and build Lotus with CPU SHA support
+ * Installed systemd service to run as user `lotus` and put logs into that home directory
+ * Used minimal stateroots snapshot @ https://fil-chain-snapshots-fallback.s3.amazonaws.com/mainnet/minimal_finality_stateroots_latest.car
+ * Started manually with `lotus daemon --import-snapshot`
+ * Stopped, then started via systemd, let it sync, watching the log output with `tail -f`
+ * Can interact with the daemon via the CLI with `lotus client` and `lotus wallet`, etc. as the `lotus` user (appears to be user-restricted IPC)
+ * Made a wallet and transferred some FIL in it for experimenting
+
+Interesting commands:
+
+ * `lotus sync wait` will run until the chain is synced, zero exit status on success
+ * `lotus wallet` - `lotus wallet new`, `lotus wallet new`, `lotus wallet balance`, `lotus wallet export` (export privkey for backup)
```

Comment:
Using local ledger with a cloud node is also a thing ( docs are lacking / pending )
https://hackmd.io/iZJacRQmSN-obAjEypEanw